corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -187,12 +190,23 @@ def get_choices_for_known_values(self, values, user):
         return []
 
     def _make_choice_from_value(self, value):
-        if value is None:
-            return Choice(NONE_CHOICE, '[Missing]')
-        elif value == "":
-            return Choice(EMPTY_CHOICE, '[Empty]')
+        if value is None or value == '':
+            return Choice(NONE_CHOICE, '[Blank]')
         return Choice(value, value)
 
+    @staticmethod
+    def _deduplicate_and_sort_choices(choices):
+        return list(sorted(DataSourceColumnChoiceProvider._deduplicate_choices(choices),
+                           key=lambda choice: choice.display))
+
+    @staticmethod
+    def _deduplicate_choices(choices):
+        found_values = set()
```

Comment:
`nit` Could be simplified to 
```
return {choice.value: choice for choice in choices}.values()
```

Reply:
clever - thanks. https://github.com/dimagi/commcare-hq/pull/27710/commits/16935183f820bb0d745bb5f54148984f70d656ad